crates/shared/src/sources/balancer_v2/pools/stable.rs

```diff
@@ -22,20 +33,164 @@ impl PoolIndexing for PoolInfo {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PoolState {
+    pub tokens: BTreeMap<H160, common::TokenState>,
+    pub swap_fee: Bfp,
```

Comment:
`swap_fee` is common to both weighted and stable pools. It was duplicated to each `PoolState` so that we can implement `BaselineSolvable` for each individual `PoolState` without requiring any additional auxiliary data.

This way, the `balancer_v2::pools::Pool` type can just contain common data that is important for interaction encoding, while each `*::PoolState` can have all the data required for computing input and output amounts.